include/lbann/layers/activations/activations.hpp

```diff
@@ -31,30 +31,52 @@
 
 namespace lbann {
 
+// Convenience macros for ETI decls for unary layers
+
+#ifndef LBANN_ACTIVATIONS_LAYER_INSTANTIATE
+#define UNARY_ETI_DECL_MACRO_DEV(LAYER_NAME, DEVICE)               \
+  extern template class entrywise_unary_layer<                     \
+    data_layout::DATA_PARALLEL, DEVICE, LAYER_NAME##_name_struct>; \
+  extern template class entrywise_unary_layer<                     \
+    data_layout::MODEL_PARALLEL, DEVICE, LAYER_NAME##_name_struct>
+#else
+#define UNARY_ETI_DECL_MACRO_DEV(...)
+#endif // LBANN_UNARY_LAYER_INSTANTIATE
+
+#ifdef LBANN_HAS_GPU
+#define UNARY_ETI_DECL_MACRO(LAYER_NAME)                       \
+  UNARY_ETI_DECL_MACRO_DEV(LAYER_NAME, El::Device::CPU);       \
+  UNARY_ETI_DECL_MACRO_DEV(LAYER_NAME, El::Device::GPU)
+#else
+#define UNARY_ETI_DECL_MACRO(LAYER_NAME)                       \
+  UNARY_ETI_DECL_MACRO_DEV(LAYER_NAME, El::Device::CPU)
+#endif // LBANN_HAS_GPU
+
 // Convenience macro to define an entry-wise unary layer class
 #define DEFINE_ENTRYWISE_UNARY_LAYER(layer_name, layer_string)          \
   struct layer_name##_name_struct {                                     \
     inline operator std::string() { return layer_string; }              \
   };                                                                    \
   template <data_layout Layout, El::Device Device>                      \
   using layer_name                                                      \
-  = entrywise_unary_layer<Layout, Device, layer_name##_name_struct>;
+  = entrywise_unary_layer<Layout, Device, layer_name##_name_struct>;    \
+  UNARY_ETI_DECL_MACRO(layer_name)
```

Comment:
Micro-quibble: When `LBANN_ACTIVATIONS_LAYER_INSTANTIATE` is active, then `UNARY_ETI_DECL_MACRO_DEV` is empty and you get two consecutive semicolons. This could cause overly pedantic compilers to throw up a warning.

Reply:
This should only happen if `-Wpedantic` or the equivalent is on. If it becomes an issue, I'll fix it.